pgjdbc/src/test/java/org/postgresql/test/jdbc42/TimestampUtilsTest.java

```diff
@@ -62,6 +63,46 @@ public void testToLocalTime() throws SQLException {
     assertEquals(LocalTime.parse("23:59:59.999999999"), timestampUtils.toLocalTime("24:00:00"));
   }
 
+  @Test
+  public void testToStringOfOffsetTime() {
+    TimestampUtils timestampUtils = createTimestampUtils();
+
+    assertEquals("00:00:00+00", timestampUtils.toString(OffsetTime.parse("00:00:00+00:00")));
+    assertEquals("00:00:00.1+01", timestampUtils.toString(OffsetTime.parse("00:00:00.1+01:00")));
+    assertEquals("00:00:00.12+12", timestampUtils.toString(OffsetTime.parse("00:00:00.12+12:00")));
+    assertEquals("00:00:00.123-01", timestampUtils.toString(OffsetTime.parse("00:00:00.123-01:00")));
+    assertEquals("00:00:00.1234-02", timestampUtils.toString(OffsetTime.parse("00:00:00.1234-02:00")));
+    assertEquals("00:00:00.12345-12", timestampUtils.toString(OffsetTime.parse("00:00:00.12345-12:00")));
+    assertEquals("00:00:00.123456+01:30", timestampUtils.toString(OffsetTime.parse("00:00:00.123456+01:30")));
+    assertEquals("00:00:00.123456-12:34", timestampUtils.toString(OffsetTime.parse("00:00:00.123456-12:34")));
+
+    assertEquals("23:59:59+01", timestampUtils.toString(OffsetTime.parse("23:59:59+01:00")));
+    assertEquals("23:59:59.999999+01", timestampUtils.toString(OffsetTime.parse("23:59:59.999999+01:00"))); // 0 NanoSeconds
+    assertEquals("23:59:59.999999+01", timestampUtils.toString(OffsetTime.parse("23:59:59.999999499+01:00"))); // 499 NanoSeconds
+    assertEquals("24:00:00+01", timestampUtils.toString(OffsetTime.parse("23:59:59.999999500+01:00")));// 500 NanoSeconds
```

Comment:
`500 NanoSeconds` should be better as an extra message parameter to the assertion.
Code comment is visible only to those who read the code, and assertion message is visible when the test fails.

Reply:
Same like before: i just duplicated already existing test and extended by timezones. The other test has same formatting.

Reply:
I can move this to the extra message param. Will do the same for the other tests in the class or should I leave it as-is to not make this PR larger?

Reply:
> should I leave it as-is to not make this PR larger?

That is up to you.
I would be ok if the old tests were left as is (e.g. to prove backward behaviour is intact)
I would be ok if the old tests were refactored to the newer approach.

I do not like adding new tests with "old" style because it makes maintenance harder. For instance, if a test fails, then it would be harder to understand why did it fail, and why a certain value was expected.

You add tests, and it is awesome. My primary ask here is to keep "newly added tests" easy to maintain (==ensure test failures look like a decent bug report).
So if you decide between "matching the old style" and "making the test comprehendable", I would almost always drop "old style", and write test in a way to make the assertion failure easy to reason.

Reply:
I fixed this test to have the rounding conditions in the message. I did this also for LocalTime.